### DISCUSSION TITLE
v0.4.0

### DISCUSSION THREAD

**Discussion by jborean93** [jborean93/PSOpenAD#72] [Announcements]
## What's Changed
* Use signing key from Azure by @jborean93 in https://github.com/jborean93/PSOpenAD/pull/64
* Update CI steps for new Azure signing key workflow by @jborean93 in https://github.com/jborean93/PSOpenAD/pull/65
* Fix up compiler warning for nullable operation by @jborean93 in https://github.com/jborean93/PSOpenAD/pull/66
* Move to ALC based build by @jborean93 in https://github.com/jborean93/PSOpenAD/pull/67
* Add cmdlets for adding and removing AD objects by @jborean93 in https://github.com/jborean93/PSOpenAD/pull/68
* Added Get-OpenADRootDSE by @jborean93 in https://github.com/jborean93/PSOpenAD/pull/69
* Prepare for v0.4.0 release by @jborean93 in https://github.com/jborean93/PSOpenAD/pull/70

## v0.4.0 - 2023-09-05

+ Moved module code into a separate Assembly Load Context to avoid assembly conflicts for dependencies
+ Added the following cmdlets:
  + [New-OpenADObject](./docs/en-US/New-OpenADObject.md): Creates new AD objects
  + [Remove-OpenADObject](./docs/en-US/Remove-OpenADObject.md): Removes AD objects
  + [Get-OpenADRootDSE](./docs/en-US/Get-OpenADRootDSE.md): Get the root directory server information tree
+ Fixed up `-LDAPFilter` logic to align the `\` escaping behaviour with OpenLDAP and the `Get-AD*` cmdlets
  + Before a filter with the char `\` had to have 2 more chars `[A-F0-9]` which represented the characters hex value
  + Now if the `\` does not have 2 characters after or they don't match the hex pattern, the `\` and subsequent values are treated literally
+ Fixed up some binary length calculations for the `PSOpenAD.Security` classes

**Full Changelog**: https://github.com/jborean93/PSOpenAD/compare/v0.3.0...v0.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/jborean93/PSOpenAD/releases/tag/v0.4.0'>v0.4.0</a>.</em>

**Comment by ravensgulls** [comment]
Hi,

This package is a good idea and works well so far! However, I see it's missing features such as enabling or disabling users/computers. Are there any plans to add this ability?

Thanks!

**Reply by jborean93** [reply]
Basic functionality for setting object has been added by https://github.com/jborean93/PSOpenAD/pull/76 and will be in the next release. For this specific scenario you can disable users/computers with the following

```powershell
# Use Get-OpenADComputer for computers
$user = Get-OpenADUser vagrant -Properties userAccountControl
$user | Set-OpenADObject -Replace @{
    userAccountControl = ($user.UserAccountControl -bor [PSOpenAD.UserAccountControl]::AccountDisable)
}
```

The reverse is as follows

```powershell
$user = Get-OpenADUser vagrant -Properties userAccountControl
$user | Set-OpenADObject -Replace @{
    userAccountControl = ($user.UserAccountControl -band -bnot [PSOpenAD.UserAccountControl]::AccountDisable)
}
```

**Reply by jborean93** [reply]
[v0.5.0](https://github.com/jborean93/PSOpenAD/discussions/78) has just been published with this new functionality.
